sympy/functions/special/singularity_functions.py

```diff
@@ -178,7 +178,33 @@ def eval(cls, variable, offset, exponent):
                 return S.Zero
             if shift.is_zero:
                 return oo
+    def __mul__(self, other):
```

Comment:
The `__mul__` method should not be defined for `Expr` subclasses because it will end up being incompatible with `Mul`.